Mathlib/Algebra/GroupPower/Lemmas.lean

```diff
@@ -2,35 +2,1345 @@
+#align zpow_left_injective zpow_left_injective
+
+@[to_additive zsmul_right_inj]
+theorem zpow_left_inj (hn : n ≠ 0) : a ^ n = b ^ n ↔ a = b :=
+  (zpow_left_injective hn).eq_iff
+#align zpow_left_inj zpow_left_inj
+
+/-- Alias of `zsmul_right_inj`, for ease of discovery alongside `zsmul_le_zsmul_iff'` and
+`zsmul_lt_zsmul_iff'`. -/
+@[to_additive
+      "Alias of `zsmul_right_inj`, for ease of discovery alongside
+      `zsmul_le_zsmul_iff'` and\n`zsmul_lt_zsmul_iff'`."]
+theorem zpow_eq_zpow_iff' (hn : n ≠ 0) : a ^ n = b ^ n ↔ a = b :=
+  zpow_left_inj hn
+#align zpow_eq_zpow_iff' zpow_eq_zpow_iff'
+
+end LinearOrderedCommGroup
+
+section LinearOrderedAddCommGroup
+
+variable [LinearOrderedAddCommGroup α] {a b : α}
+
+theorem abs_nsmul (n : ℕ) (a : α) : |n • a| = n • |a| := by
+  cases' le_total a 0 with hneg hpos
+  · rw [abs_of_nonpos hneg, ← abs_neg, ← neg_nsmul, abs_of_nonneg]
+    exact nsmul_nonneg (neg_nonneg.mpr hneg) n
+  · rw [abs_of_nonneg hpos, abs_of_nonneg]
+    exact nsmul_nonneg hpos n
+#align abs_nsmul abs_nsmul
+
+theorem abs_zsmul (n : ℤ) (a : α) : |n • a| = |n| • |a| := by
+  obtain n0 | n0 := le_total 0 n
+  · obtain ⟨n, rfl⟩ := Int.eq_ofNat_of_zero_le n0
+    simp only [abs_nsmul, coe_nat_zsmul, Nat.abs_cast]
+  · obtain ⟨m, h⟩ := Int.eq_ofNat_of_zero_le (neg_nonneg.2 n0)
+    rw [← abs_neg, ← neg_zsmul, ← abs_neg n, h, coe_nat_zsmul, Nat.abs_cast, coe_nat_zsmul]
+    exact abs_nsmul m _
+#align abs_zsmul abs_zsmul
+
+theorem abs_add_eq_add_abs_le (hle : a ≤ b) : |a + b| = |a| + |b| ↔ 0 ≤ a ∧ 0 ≤ b ∨ a ≤ 0 ∧ b ≤ 0 :=
+  by
+  obtain a0 | a0 := le_or_lt 0 a <;> obtain b0 | b0 := le_or_lt 0 b
+  · simp [a0, b0, abs_of_nonneg, add_nonneg a0 b0]
+  · exact (lt_irrefl (0 : α) <| a0.trans_lt <| hle.trans_lt b0).elim
+  any_goals simp [a0.le, b0.le, abs_of_nonpos, add_nonpos, add_comm]
+  have : (|a + b| = -a + b ↔ b ≤ 0) ↔ (|a + b| = |a| + |b| ↔ 0 ≤ a ∧ 0 ≤ b ∨ a ≤ 0 ∧ b ≤ 0) := by
+    simp [a0, a0.le, a0.not_le, b0, abs_of_neg, abs_of_nonneg]
+  refine' this.mp ⟨fun h => _, fun h => by simp only [le_antisymm h b0, abs_of_neg a0, add_zero]⟩
+  obtain ab | ab := le_or_lt (a + b) 0
+  · refine' le_of_eq (eq_zero_of_neg_eq _)
+    rwa [abs_of_nonpos ab, neg_add_rev, add_comm, add_right_inj] at h
+  · refine' (lt_irrefl (0 : α) _).elim
+    rw [abs_of_pos ab, add_left_inj] at h
+    rwa [eq_zero_of_neg_eq h.symm] at a0
+#align abs_add_eq_add_abs_le abs_add_eq_add_abs_le
+
+theorem abs_add_eq_add_abs_iff (a b : α) : |a + b| = |a| + |b| ↔ 0 ≤ a ∧ 0 ≤ b ∨ a ≤ 0 ∧ b ≤ 0 := by
+  obtain ab | ab := le_total a b
+  · exact abs_add_eq_add_abs_le ab
+  · rw [add_comm a, add_comm (abs _), abs_add_eq_add_abs_le ab, and_comm, @and_comm (b ≤ 0)]
+#align abs_add_eq_add_abs_iff abs_add_eq_add_abs_iff
+
+end LinearOrderedAddCommGroup
+
+-- Porting note: commented out as linter reports LHS= RHS syntactically
+-- @[simp]
+-- theorem WithBot.coe_nsmul [AddMonoid A] (a : A) (n : ℕ) : ((n • a : A) : WithBot A) = n • a :=
+--   by simp
```

Comment:
Actually the linter is not telling you to comment it out, but rather to fix the statement:
```
@[simp]
theorem WithBot.coe_nsmul [AddMonoid A] (a : A) (n : ℕ) : (n • a : A) = n • (a : WithBot A) :=
  AddMonoidHom.map_nsmul
    { toFun := fun a : A => (a : WithBot A), 
      map_zero' := WithBot.coe_zero, 
      map_add' := WithBot.coe_add }
    a n
```

Reply:
Probably the similar one below needs the same treatment.

Reply:
Thanks. Fixed one. For the other I will wait for @riccardobrasca as it involves his work.

Reply:
Fixed here and below.